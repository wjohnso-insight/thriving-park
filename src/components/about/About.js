//@flow
import * as React from 'react'
import axios from 'axios'

import './About.scoped.scss'

type Props = {

}

export default function About(props: Props) : React.Node {
    
    const { useState, useEffect } = React;
    const [ feedPosts, setFeedPosts] = useState()

    useEffect(() => {
        axios.get('api/feedPosts')
        .then((res) => setFeedPosts(res.data.feedPosts))
        .catch((err) => console.error(err))
    },[])

    const feedPostsRenderer = () => {
        if(feedPosts){
            return(
                feedPosts.map(post => <span key={post.id}>{post.body}</span>)
            )
        }else{
            return <span>Getting posts</span>
        }
    }
    
    return (
        <div className="content" id="about-wrapper">
            <section className="content-panel" id="community-feed">
                <h1 className="graf sign" data-testid="feed-header">Community Feed</h1>
                {feedPostsRenderer()}
            </section>
        </div>
    )
}