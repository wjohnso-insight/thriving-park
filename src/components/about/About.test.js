import React from 'react'
import About from './About'
import { render, cleanup, getByTestId } from '@testing-library/react'

describe('describes the About component', () =>{

    beforeEach(() => {
        console.error = jest.fn();
        console.error('Error')
    })

    afterEach(() => {
        cleanup()
    })

    it('should render the component', () =>{
        const component = render(<About />)
        expect(component).toMatchSnapshot();
    })

    it('should render the Community Feed', () => {
        const { container } = render(<About />);
        const feedHeader = getByTestId(container, 'feed-header')
        expect(feedHeader).toBeInTheDocument();
    })

})