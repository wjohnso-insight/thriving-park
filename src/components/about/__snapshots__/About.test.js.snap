// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describes the About component should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content"
        id="about-wrapper"
      >
        <section
          class="content-panel"
          id="community-feed"
        >
          <h1
            class="graf sign"
            data-testid="feed-header"
          >
            Community Feed
          </h1>
          <span>
            Getting posts
          </span>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content"
      id="about-wrapper"
    >
      <section
        class="content-panel"
        id="community-feed"
      >
        <h1
          class="graf sign"
          data-testid="feed-header"
        >
          Community Feed
        </h1>
        <span>
          Getting posts
        </span>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
